### DISCUSSION TITLE
Parse and evaluate dynamic formula

### DISCUSSION THREAD

**Discussion by binajmen** [supabase/supabase#3833] [Questions]
Hi everyone,

This is clearly a pure Postgres question rather than Supabase specific. I didn't know where to ask such a question. The Supabase community being super friendly and planning on building a new project on top of it.. The heck, why not ask here!

Having a table `Layer`:
| id | name | formula |
|-|-|-|
| 1 | rectangle | width * height |

And a table `Variables`:
| id | layer_id | name | value |
|-|-|-|-|
| 1 | 1 | width | 2 |
| 2 | 1 | height | 4 |

Is it possible to build a view `Layer_Computed` with pure Postgres?
| layer_id | result |
|-|-|
| 1 | 8 |

The "tricky" part would be to parse the formula, replace with the values, and evaluate the whole.

An example is of course more than welcome, but having pointers to some docs/right terms to search if also fine :)

**Comment by steve-chavez** [comment]
You could use `SELECT regexp_split_to_array('width * height', '\s+');` to parse your input and then work with the array.

But plpgsql has `EXECUTE`, so you could just wrap that into an `eval` function like done [here](https://stackoverflow.com/a/16058077/4692662).

If the `formula` column will be inserted by external users, then you'd probably want to be strict and use the first option, I'd avoid an `eval` in that case.
